app/models/ability.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user = nil)
+    if user
+      can :manage, Move, Move.served_by(user)
+    else
+      can :manage, Move
+    end
+  end
+end
```

Comment:
This needs tests, so we can build on them as it becomes more complex. Something like https://github.com/CanCanCommunity/cancancan/wiki/Testing-Abilities#rspec would be great.

Reply:
If we're introducing CanCanCan as an authorisation layer, we should plan to abstract the Doorkeeper authorisation checks into here. With that in mind, could we have a default of "can do nothing"? As an example, in the above we could receive the Doorkeeper application and only give the `:manage` ability to authenticated users, scoping it down if the supplier is present.

Reply:
Correct me if I'm wrong: the doorkeeper authentication is in the `api_controller`, if someone is unauthenticated it won't be able to do anything and will never get to `cancancan`, which will evaluate `current_user` only AFTER authentication, right?

Reply:
That is correct, but it makes me a bit nervous to split our authorisation scheme across two mechanisms as it's easy to see a flaw being introduced in future, e.g. if we introduced unauthenticated endpoints. If we build this ability assuming nothing it should be easier to test, and describes our authorisation rules to a new member of the team without needing to dig through other parts of the system